elastic/elastic-agent
# Add Elastic Infra Metrics processor to `otel` mode

## What does this PR do?

This PR makes the [Elastic Infra Metrics processor](https://github.com/elastic/opentelemetry-collector-components/tree/processor/elasticinframetricsprocessor/v0.1.0/processor/elasticinframetricsprocessor) available to users running Elastic Agent in `otel` mode.

## Why is it important?

The processor translates OTel native metrics to those understood by the Elastic system integration.

## Checklist


## Disruptive User Impact

No impact expected.

## How to test this PR locally

1. Build the agent locally (see [docs](https://github.com/elastic/elastic-agent/blob/main/README.md#packaging))
1. Unpack the .tar.gz archive in build/distributions
1. Run the built binary with `build/distributions/elastic-agent-*/elastic-agent otel --config internal/pkg/otel/testdata/all-components.yml

## Related issues

- https://github.com/elastic/elastic-agent/issues/4969
